app/lib/mets_directory_scanner.rb

```diff
@@ -0,0 +1,72 @@
+# frozen_string_literal: true
+require 'find'
+
+# performs scans on directories for XML files and processes them if they are not already done or in progress
+class MetsDirectoryScanner
+  def self.perform_scan
+    MetsDirectoryScanner.scan_directories.each do |directory|
+      Find.find(directory) do |path|
+        next unless path =~ /.*_mets\.xml$/ && path !~ /.*xslt_result_mets\.xml$/
+        is_done = File.exist?(File.join(File.dirname(path), "#{indicator_file_prefix}.done"))
+        progress_file = File.join(File.dirname(path), "#{indicator_file_prefix}.progress")
+        File.delete(progress_file) if File.exist?(progress_file) && File.mtime(progress_file) < (Time.now.utc - 1.day)
+        is_in_progress = File.exist?(progress_file)
+        begin
+          unless is_done || is_in_progress
+            #  The following will fail if file is not created by this call:
+            IO.sysopen(progress_file, Fcntl::O_WRONLY | Fcntl::O_EXCL | Fcntl::O_CREAT)
```

Comment:
I'm not familiar with these options - would it be worth it to add a little bit more about them, or a link to documentation? Not required change, just a thought.